pkg/cloud/aws/actuators/machine_scope.go

```diff
@@ -117,9 +120,10 @@ func (m *MachineScope) GetScope() *Scope {
 
 // Close the MachineScope by updating the machine spec, machine status.
 func (m *MachineScope) Close() {
-	if m.MachineClient == nil {
+	if m.MachineClient == nil || m.Machine == nil {
```

Comment:
What would cause `m.Machine` to be nil? Wouldn't we always have a Machine?

Reply:
heh, I think this snuck in. I'll remove it and cluster nil check.